src/main/java/org/seattlevoluntech/controllers/ProjectController.java

```diff
@@ -86,6 +88,53 @@ public Project createProject(@RequestBody Project project){
         return projectsRepository.findById(id);
     }
 
+    // Update project
+    @PutMapping(path="/projects/link/{projectId}/{userId}")
+    @ResponseBody
+    public Optional<Project> linkUserWithProject(
+            @PathVariable("projectId") Long projectId,
+            @PathVariable("userId") Long userId
+    ) {
+        Optional<Project> optionalProject = projectsRepository.findById(projectId);
+
+        if(!optionalProject.isPresent())
+        {
+            throw new ErrorController.ErrorNotFound("projectId-" + projectId);
+        }
+
+        Optional<User> optionalUser = usersRepository.findById(userId);
+
+        if (!optionalUser.isPresent())
+        {
+            throw new ErrorController.ErrorNotFound("userId-" + userId);
+        }
+
+
+        optionalProject.get().addUsers(optionalUser.get());
+        optionalUser.get().addProject(optionalProject.get());
+
+        projectsRepository.save(optionalProject.get());
+        usersRepository.save(optionalUser.get());
```

Comment:
Do we need to make sure the updates to projects and users tables are atomic?

Reply:
That's a good point. I'm going to assume that we read more than write and use optimistic locking here. Let me know if you have any objections.

Reply:
Added optimistic locking on both user and project tables. However when manually inserting test data you have to ensure that the opt_lock field is filled with a integer.